HelpSource/Classes/ArrayedCollection.schelp

```diff
@@ -439,25 +463,40 @@ a.performInPlace(\normalizeSum, 3, 6);
 ::
 
 method::rank
-Rank is the number of dimensions in a multidimensional array.
+Requires link::#-isRectangular:: as a precondition.
+Returns the number of dimensions the value has.
+A single value (scalar) has rank of 0, while an empty array has rank of 1.
+If the array is not rectangular the wrong result will be returned, as this only considers the left-most elements.
+
+Optionally, rank takes an boolean as an argument indicating if it should test the array is rectangular.
+
 code::
-a = [4,7,6,8];
-a.rank;
-a = [[4,7],[6,8]];
-a.rank;
+[].rank // 1
+[4,7,6,8].rank // 1
+[[4,7],[6,8]].rank // 2
+[1 ,2, [3, 4]].rank // 1, this array is not rectangular and returns a meaningless value.
+[1 ,2, [3, 4]].rank(test: true) // throws an error
 ::
 
+
+
 method::shape
-For a multidimensional array, returns the number of elements along each dimension.
+Requires link::#-isRectangular:: as a precondition.
+For a multidimensional array, returns an array of the number of elements along each consecutive dimension.
+If the array is not rectangular the wrong result will be returned, as this only considers nested arrays at index zero.
+
+Optionally, shape takes an boolean as an argument indicating if it should test the array is rectangular.
```

Comment:
See my suggested structuring of `rank` above

Reply:
Okay I think that is better